SecureImage/UI/Albums/AlbumsViewController.swift

```diff
@@ -34,39 +43,137 @@ class AlbumsViewController: UIViewController {
     }()
     
     override func viewDidLoad() {
-        super.viewDidLoad()
 
+        super.viewDidLoad()
+        
         commonInit()
     }
-
-    override func didReceiveMemoryWarning() {
-        super.didReceiveMemoryWarning()
+    
+    override func viewWillAppear(_ animated: Bool) {
+    
+        super.viewWillAppear(animated)
+        
+        do {
+            albums = try Realm().objects(Album.self)
+            tableView.reloadData()
+        } catch {
+            print("Unable to load Albums")
+        }
     }
     
     override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+
          let dvc = segue.destination as! AlbumDetailsViewController
-         dvc.localAlbumID  = "11111111111"
+         dvc.localAlbumID  = selectedAlbumID
     }
     
     private func commonInit() {
-        tableView.isHidden = true
-        
+
         view.addSubview(createFirstAlbumView)
         
+        tableView.dataSource = self
+        tableView.delegate = self
+
+        createAlbumButton.layer.cornerRadius = createAlbumButton.frame.height / 2
+        createAlbumButton.clipsToBounds = true
+        
         createFirstAlbumView.widthAnchor.constraint(equalToConstant: 250).isActive = true
         createFirstAlbumView.heightAnchor.constraint(equalToConstant: 100).isActive = true
         createFirstAlbumView.centerXAnchor.constraint(equalTo: view.centerXAnchor).isActive = true
         createFirstAlbumView.centerYAnchor.constraint(equalTo: view.centerYAnchor).isActive = true
         
         createFirstAlbumView.onCreateFirstAlbumTouched = { [weak self] in
-            // Create a new album in the db
-            // Pass album ID to Album Details VC
+            self?.createAlbumTouched(sender: nil)
+        }
+    }
+    
+    private func configureFor(albumsExist: Bool) {
+        
+        if albumsExist {
```

Comment:
an if - else would be cleaner:

if albumsExist {
	tableView.isHidden = false
	createFirstAlbumView.isHidden = true
	createAlbumButton.isHidden = false
} else {
	tableView.isHidden = true
	createFirstAlbumView.isHidden = false
	createAlbumButton.isHidden = true
}

Reply:
I think that, in general, early returns should be favoured over conditional branches because they get conditions out of the way early and they also make the logic easier to follow.